File: 20221102/모범답안/20221102_04/JoinToString.py
def solution(arr, prefix, separator, postfix):
    """
    :param arr: int[]
    :param prefix: string
    :param separator: postfix
    :param postfix: postfix
    :return: postfix
    """

    result = ''

    if prefix is not None:
        result += prefix

    for i, val in enumerate(arr):
        if i > 0:
            result += separator

        result += str(val)

    if postfix is not None:
        result += postfix

    return result

File: 20221102/모범답안/20221102_04/JoinToString.js
/**
 * @param arr {number[]}
 * @param prefix {string}
 * @param separator {string}
 * @param postfix {string}
 * @returns {string}
 */
function solution(arr, prefix, separator, postfix) {
  let result = ""

  if (prefix !== undefined) {
    result += prefix
  }

  for (const [i, v] of arr.entries()) {
    if (i > 0) {
      result += separator
    }
    result += v
  }

  if (postfix !== undefined) {
    result += postfix
  }

  console.log(result)
  return result
}

export default solution
